test_programs/compile_success_empty/comptime_function_definition/src/main.nr

```diff
@@ -0,0 +1,31 @@
+use std::meta::type_of;
+
+struct Foo { x: Field }
+
+#[function_attr]
+fn foo(w: i32, y: Field, Foo { x }: Foo, mut a: bool) -> i32 {
```

Comment:
Might as well expand `Foo { x }` a bit to also check we handle multiple struct fields correctly with only some fields being shortened. E.g. `Foo { x, y: foo_y }`